src/core/apollo/hooks/useApolloErrorHandler.ts

```diff
@@ -1,21 +1,31 @@
 import { ApolloError } from '@apollo/client';
 import { GraphQLError } from 'graphql';
-import { Severity } from '../../state/global/notifications/notificationMachine';
+import { i18n, TFunction } from 'i18next';
 import { useTranslation } from 'react-i18next';
+import { Severity } from '../../state/global/notifications/notificationMachine';
 import { error as logError } from '../../../services/logging/sentry/log';
 import { useNotification } from '../../ui/notifications/useNotification';
 
-enum GraphQLErrorsExtensionCodes {
-  BAD_USER_INPUT = 'BAD_USER_INPUT',
-}
-
-const tryGetField = (errorMessage: string): string | undefined => {
-  let matches = errorMessage?.match(/property ([\w-]+) has failed/);
-  return matches ? matches[1] : undefined;
+const getTranslationForCode = (error: GraphQLError, t: TFunction, i18n: i18n) => {
+  const code = error.extensions?.code as string;
+  if (!code) {
+    // if code missing send a generic error text
+    return t('apollo.errors.generic');
+  }
+
+  const key = `apollo.errors.${code}`;
+
+  if (!i18n.exists(key)) {
+    // if the error text is missing for that code
+    // send a generic error text with code
+    return t('apollo.errors.generic-with-code', { code });
```

Comment:
Unfortunately, I don't believe it will in a consistent way, because we don't have a policy for the message content and I don't want to expose the raw exception message to the user.
If you have any ideas as to what more I can do with the message, please share.